src/compas_model/elements/element.py

```diff
@@ -752,228 +430,324 @@ def oobb_mesh(self):
+                ]
+                count += 1
 
-        # compute axis-aligned-bounding-box of all objects
-        points_bbox = bounding_box(points_bbox)
+        # --------------------------------------------------------------------------
+        # return the avarege of all centroids
+        # --------------------------------------------------------------------------
+        self._center_of_mass = [sum_of_points[i] / count for i in range(3)]
+        return self._center_of_mass
 
-        # inflate the box
-        self._aabb = bounding_box(
-            [
-                [
-                    points_bbox[0][0] - inflate,
-                    points_bbox[0][1] - inflate,
-                    points_bbox[0][2] - inflate,
-                ],
-                [
-                    points_bbox[6][0] + inflate,
-                    points_bbox[6][1] + inflate,
-                    points_bbox[6][2] + inflate,
-                ],
-            ]
-        )
+    @property
+    def centroid(self):
+        x_sum = 0.0
+        y_sum = 0.0
+        z_sum = 0.0
+        count = 0
+
+        for g in self.geometry:
+            # --------------------------------------------------------------------------
+            # get centroid from all meshes
+            # --------------------------------------------------------------------------
+            if isinstance(g, Mesh):
+                x, y, z = centroid_points(
+                    [g.vertex_coordinates(key) for key in g.vertices()]
+                )
+                x_sum += x
+                y_sum += y
+                z_sum += z
+                count += 1
 
-        # return the aabb (8 points)
-        return self._aabb
+        # --------------------------------------------------------------------------
+        # return the avarege of all centroids
+        # --------------------------------------------------------------------------
+        self._centroid = Point(x_sum / count, y_sum / count, z_sum / count)
+        return self._centroid
 
-    def oobb(self, inflate=0.00):
-        """Compute the oriented bounding box based on geometry geometries points"""
-        # define this property dynamically in the class
-        if not hasattr(self, "_oobb"):
-            self._oobb = []  # XYZ coordinates of 8 points defining a box
+    @property
+    def is_support(self):
+        if not hasattr(self, "_is_support"):
+            self._is_support = False
+        return self._is_support
 
-        # if the oobb is already computed return it
-        # and the user does not give another inflation value
-        if self._oobb and abs(inflate) < 0.001:
-            return self._oobb
+    @is_support.setter
+    def is_support(self, value):
+        self._is_support = value
 
-        # iterate geometry and get the bounding box by geometry name
-        # Mesh, Polyline, Box, Line
-        points = []
+    @property
+    def display_schema(self):
```

Comment:
?

Reply:
Will be replaced following the Scene representation.
This method is used to specify which and how properties are vizualized.

Reply:
Since I do not know if the Scene implementation is fully finished. 
Here is an explanation what I am trying to do here:

1) Structural Elements: Beams, Blocks, Plates, Interfaces and etc. are essentially groups of geometrical properties with very minimal functionality.
2) This means you usually want to check for example the polygon winding direction, the beam center points, the plate frame.
3) For this I often was doing a user interface to toggle ON/OFF different attributes. Handy for debugging when you are developing some sort of algorithm for a certain project.

Please refer to the image below:
![Artboard 1@4x](https://github.com/BRG-research/compas_model/assets/18013985/c0dbbc1d-7884-4a9a-849a-e6a19eaefcd0)